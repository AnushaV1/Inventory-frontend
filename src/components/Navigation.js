import React from "react";
import { NavLink } from "react-router-dom";
import { useSelector, useDispatch} from 'react-redux';
import { useHistory } from 'react-router-dom';
import homeInventoryLogo from "../images/homeInventoryLogo.jpg";
import { logoutUser } from "../actions/userAction";
import Search from "./Search";
import './Navigation.css';


    const Navigation = () => {
        const currentUser = useSelector(st => st.users.user);
        const dispatch = useDispatch();
        const history = useHistory();

        const logout = async() => {
            await dispatch(logoutUser());
            history.push("/")
        }

    return (
    <>
    <nav className="navbar navbar-expand-md navbar-dark bg-dark">
    <div className="container-fluid">
    <div className="navbar-header">
    <NavLink to="/" className="navbar-brand">
    <img src={homeInventoryLogo} alt="Home Inventory Logo"/>My Home Inventory
    </NavLink>
    </div>    
        {currentUser.username ? ( 
            <>
            <Search />
    
        <ul className="navbar-nav navbar-right"> 
        <li className="nav-item">
        <NavLink className="nav-link" to="/dashboard">Dashboard </NavLink>
        </li>
        <li className="nav-item">
        <NavLink className="nav-link" to="/profile">Profile </NavLink>
        </li>
        <li className="nav-item">
        <NavLink className="nav-link" to="/addItem">Add Product </NavLink>
        </li>
        <li className="nav-item">
            <NavLink className="nav-link" to='/login' onClick={logout}>Logout</NavLink>
        </li>
        </ul>        
        </>
        ) : (
            <>
        <ul className="nav navbar-nav navbar-right"> 
        <li className="nav-item"><NavLink className="nav-link" to="/register">Register</NavLink></li>
        <li className="nav-item"><NavLink className="nav-link" to="/login">Login</NavLink></li>
        </ul>
        </> ) 
        }
        </div>
    </nav>
    </>
    )
}

export default Navigation;