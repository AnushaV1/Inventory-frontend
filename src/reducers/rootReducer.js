import { combineReducers } from 'redux';
import { users }  from "./authReducer";
import  { alert }  from "./alertReducer";
import  { upc }  from "./upcReducer";
import { products } from "./userInventoryReducer"

const rootReducer = combineReducers({

    users,
    upc,
    alert,
    products
});

export default rootReducer;