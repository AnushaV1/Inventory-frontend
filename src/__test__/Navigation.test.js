import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import rootReducer from '../reducers/rootReducer';
import { BrowserRouter } from 'react-router-dom';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import Navigation from '../components/Navigation';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));

it('renders without crashing', function () {
	render(
		<Provider store={store}>
			<BrowserRouter>
				<Navigation />
			</BrowserRouter>
		</Provider>
	);
});

it('matches snapshot', function () {
	const { asFragment } = render(
		<Provider store={store}>
			<BrowserRouter>
				<Navigation />
			</BrowserRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});