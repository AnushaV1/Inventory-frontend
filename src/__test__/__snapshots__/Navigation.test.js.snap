// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <div
      class="container-fluid"
    >
      <div
        class="navbar-header"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          <img
            alt="Home Inventory Logo"
            src="homeInventoryLogo.jpg"
          />
          My Home Inventory
        </a>
      </div>
      <ul
        class="nav navbar-nav navbar-right"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/register"
          >
            Register
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
